docs/references/rest-apis/rest-service-listing-api.md

```diff
@@ -1,144 +1,332 @@
+  * **REST API path** : /services/serviceListing/v1/factoryPids/byFilter
+  * **description** : Returns the list of factory pids whose properties match the specified filter.
+  * **request body** :
+      * [Filter](#filter)
+  * **responses** :
+      * **200**
+          * **description** : An object reporting the matching factory pids.
+          * **response body** :
+              * [PidSet](#pidset)
+      * **400**
+          * **description** : If the request body is not syntactically correct.
+          * **response body** :
+              * [GenericFailureReport](#genericfailurereport)
+      * **500**
+          * **description** : If an unexpected failure occurs while retrieving the factory pid list.
+          * **response body** :
+              * [GenericFailureReport](#genericfailurereport)
 
-```JSON
-Filter example
+## JSON definitions
+### InterfaceNames
+A list of serviceinterface names.
 
+<br>**Properties**:
+
+  * **interfaceNames**: `array` 
+      The list of service interface names.
+
+      * array element type: `string`
+          A service interface name.
+
+  
+
+```json
 {
-  "interfacesIds": [
-    "org.eclipse.kura.security.keystore.KeystoreService"
+  "interfaceNames": [
+    "org.eclipse.kura.security.keystore.KeystoreService",
+    "org.eclipse.kura.configuration.ConfigurableComponent"
   ]
 }
-
 ```
+### Reference
+A object specifying a service pid and a reference name.
+
+<br>**Properties**:
+
+  * **pid**: `string` 
+      The pid of the service containing the reference
 
-```JSON
+  
+  * **referenceName**: `string` 
+      The reference name
+
+  
+
+```json
 {
-    //Alphabetically ordered list of services running on kura filtered by implemented interfaces
-	"servicesList": [
-        "HttpsKeystore",
-        "SSLKeystore"
-    ]
+  "pid": "org.eclipse.kura.ssl.SslManagerService",
+  "referenceName": "KeystoreService"
 }
 ```
+### PropertyFilter
+A filter matching property keys and values. If the `value` property omitted the filter will match if the property is set, regardless of the value. If the service property value is of array or list type, the filter will match if at least one of the elements match.
+
+<br>**Properties**:
+
+  * **name**: `string` 
+      The property name that should be matched, it must not contain spaces
 
-If more than one interface is present in the `interfaceIds` request body, a logic AND of those are performed.
-Example:
+  
+  * **value**: `string` (**optional**)
+      The property value that should be matched.
 
-```JSON
-Filter example
+  
 
+```json
 {
-  "interfacesIds": [
-    "org.eclipse.kura.configuration.ConfigurableComponent",
-    "org.eclipse.kura.watchdog.WatchdogService"
-  ]
+  "name": "foo",
+  "value": "bar"
 }
 ```
-```JSON
-Response:
+```json
 {
-    "servicesList": [
-        "org.eclipse.kura.watchdog.WatchdogService"
-    ]
+  "name": "foo"
 }
 ```
+### NotFilter
+A filter that matches if filter specified by the "not" propertiy does not match.
```

Comment:
> A filter that matches if filter specified by the "not" propertiy does not match.

Successfully returning if the filter specified by the "not" property does not match.